### ISSUE TITLE
Towards a nicer Makie recipe

### ISSUE THREAD

**Issue by asinghvi17** [rafaqz/DimensionalData.jl#720]
It occurs to me that dimensional axes are _dimensional_, and we can use Makie's new unit integration to deal with them.

For example, instead of defining conversions in convert_arguments, we might also define `expand_dimensions` for dimensional arrays, so that Makie can pick up on the dimension of the unit.  Defining a custom dim conversion might also allow axis modifications like default labels, without the need to override the basic `plot!` function.

This isn't particularly urgent, but would be (a) nice to have and (b) pave the way for more such integration in the Makie ecosystem.

**Comment by asinghvi17** [comment]
I just ran into an issue trying to plot a Raster using GeoMakie :D 

Consider the following:
```julia
using RasterDataSources, Rasters, ArchGDAL, CairoMakie, GeoMakie
ras = Raster(EarthEnv{HabitatHeterogeneity}, :homogeneity) # habitat homogeneity to neighbouring pixel
surface(ras[:homogeneity]; axis = (; type = GeoAxis))
```

This causes an error:
```julia
ERROR: MethodError: no method matching setindex!(::Nothing, ::Tuple{String, String}, ::Symbol, ::Symbol)
Stacktrace:
 [1] surface(A::Raster{…}; x::Nothing, y::Nothing, colorbarkw::@NamedTuple{}, attributes::@Kwargs{…})
   @ DimensionalDataMakie ~/.julia/dev/DimensionalData/ext/DimensionalDataMakie.jl:143
 [2] top-level scope
   @ ~/.julia/dev/GeoMakie/examples/rasters.jl:13
Some type information was truncated. Use `show(err)` to see complete types.
```

because the plotting code assumes that surface is always plotted in a LScene, and more importantly it doesn't query what kind of axis has been created.  This can be solved independently, but it's something which I feel could be solved more cleanly with this approach.

**Comment by rafaqz** [comment]
This all sounds good to me. But there are potential problems like: we add a `Colorbar` to some plots, set axis names manually, and add our own attributes to `Attributes`.

I think we will need an example PR to show if this is viable or not, I suspect currently its not.

**Comment by asinghvi17** [comment]
FYI, 
```julia
function Makie.expand_dimensions(::Makie.ImageLike, raster::AbstractRaster)
    return Makie.convert_arguments(Makie.ImageLike(), raster)
end
```
just works.  I ran into this issue with GeoMakie's meshimage recently while trying to plot a Raster through it.


We can also avoid evaling things and go through `create_plot` once  https://github.com/MakieOrg/Makie.jl/pull/3974 lands.

**Comment by rafaqz** [comment]
Are you saying we just need to add that method?

I don't totally follow

**Comment by asinghvi17** [comment]
Yeah we just need to add expand dimensions for each trait to make DimArray handling generic across traits!

**Comment by asinghvi17** [comment]
https://github.com/rafaqz/DimensionalData.jl/blob/58dedd6f6a6381f9882bbb59981972be72d73763/ext/DimensionalDataMakie.jl#L353-L357 seems like piracy if I understand correctly?

**Comment by rafaqz** [comment]
Ah should probably be AbstractDimMatrix?

Maybe a typo

**Comment by asinghvi17** [comment]
I'll fix in the PR and see if anything breaks :D

**Comment by rafaqz** [comment]
Ok beat me to it

**Comment by asinghvi17** [comment]
Unfortunately #747 doesn't seem to solve the issue of dimensional axes.  I can't actually see where the stack overflow is happening though.
